### ISSUE TITLE
Can this method be applied to equations containing elements with a mixture of time and space derivatives?

### ISSUE THREAD

**Issue by 111aaabbb** [isds-neu/PhyCRNet#19]
Hello,

Thanks a lot for this great work that you published transparently.

I have a question that  can this method be applied to equations containing elements with a mixture of time and space derivatives? such like Ut+Utt+Uxxt+Uyyt=0.

And in the code, ' output = torch.cat((output[:, :, :, -1:], output, output[:, :, :, 0:2]), dim=3) ', why are the first（output[:, :, :, -2:]） and last（output[:, :, :, 0:3]） elements not symmetrical? ( why not-2: and 0:2 )

Thank you very much for your answer, looking forward to your reply！

**Comment by paulpuren** [comment]
> Hello,
> 
> Thanks a lot for this great work that you published transparently.
> 
> I have a question that can this method be applied to equations containing elements with a mixture of time and space derivatives? such like Ut+Utt+Uxxt+Uyyt=0.
> 
> And in the code, ' output = torch.cat((output[:, :, :, -1:], output, output[:, :, :, 0:2]), dim=3) ', why are the first（output[:, :, :, -2:]） and last（output[:, :, :, 0:3]） elements not symmetrical? ( why not-2: and 0:2 )
> 
> Thank you very much for your answer, looking forward to your reply！

Hello, thanks for your interest. I think that should be fine for a mixture of time and space derivatives. Our method is applicable to the general setup of finite difference-based schemes. For example, you can do U_xx first and then compute U_xxt based on U_xx.

For the padding dimension issues, that is because the spatial dimension of the solution is 129x129, I guess.

**Comment by 111aaabbb** [comment]
> > Hello,
> > Thanks a lot for this great work that you published transparently.
> > I have a question that can this method be applied to equations containing elements with a mixture of time and space derivatives? such like Ut+Utt+Uxxt+Uyyt=0.
> > And in the code, ' output = torch.cat((output[:, :, :, -1:], output, output[:, :, :, 0:2]), dim=3) ', why are the first（output[:, :, :, -2:]） and last（output[:, :, :, 0:3]） elements not symmetrical? ( why not-2: and 0:2 )
> > Thank you very much for your answer, looking forward to your reply！
> 
> Hello, thanks for your interest. I think that should be fine for a mixture of time and space derivatives. Our method is applicable to the general setup of finite difference-based schemes. For example, you can do U_xx first and then compute U_xxt based on U_xx.
> 
> For the padding dimension issues, that is because the spatial dimension of the solution is 129x129, I guess.

Thank you very much for your answer, I will try it again!
